modules/core/src/main/java/org/apache/fluo/core/oracle/OracleServer.java

```diff
@@ -410,15 +417,18 @@ public void takeLeadership(CuratorFramework curatorFramework) throws Exception {
       }
 
       synchronized (this) {
+        // FLUO-1000
+        isLeader = true;
+        while (!curatorFramework.getState().equals(CuratorFrameworkState.STARTED)) {
```

Comment:
Would it make sense to move this loop the `OracleServer.start()` method after `curatorFramework.start()` is called?

Reply:
Tried that, it failed. @keith-turner Thank you for looking at this, I've put a lot of invisible work into it and am greatful for your review.

Reply:
Could take another look at that, and try it again though because I did it that way because likewise that makes the best sense.